test/jackdaw/streams_test.clj

```diff
@@ -475,6 +470,41 @@
         (is (= [2 3] (second keyvals)))
         (is (= [2 7] (nth keyvals 2))))))
 
+  (testing "flat-transform"
+    (let [topic-a (mock/topic "topic-a")
+          topic-b (mock/topic "topic-b")
+          transformer-supplier-fn #(let [total (atom 0)]
+                                     (reify Transformer
+                                       (init [_ _])
+                                       (close [_])
+                                       (transform [_ k v]
+                                         ;; each input creates two outputs
+                                         ;; each v' accumulating the v read in:
+                                         ;; [[k * 10, v'] [k * 20, v'']]
+                                         (map (fn [x]
+                                                (swap! total + v)
+                                                (key-value [(* k x) @total]))
+                                              [10 20]))))
+          driver (mock/build-driver (fn [builder]
+                                      (-> builder
+                                          (k/kstream topic-a)
+                                          (k/flat-transform transformer-supplier-fn)
+                                          (k/to topic-b))))
+          publish (partial mock/publish driver topic-a)]
+
+      (publish 1 1)
+      (publish 1 2)
+      (publish 1 4)
+
+      (let [keyvals (mock/get-keyvals driver topic-b)]
+        (is (= 6 (count keyvals)))
```

Comment:
That's quite verbose, why not asserting the whole value of the data structure at once?

Reply:
Personally I find it easier to debug when one part of the seq goes wrong (you're not trying to work out yourself which part is incorrect). Although granted in this example it's likely to more spectacularly break in a regression.